internal/service/fsx/lustre_file_system.go

```diff
@@ -510,14 +509,18 @@ func resourceLustreFileSystemUpdate(ctx context.Context, d *schema.ResourceData,
 			input.LustreConfiguration.DailyAutomaticBackupStartTime = aws.String(d.Get("daily_automatic_backup_start_time").(string))
 		}
 
-		if v, ok := d.GetOk("data_compression_type"); ok {
```

Comment:
Can you elaborate on why you modified this attributes behavior?

Reply:
Couple of reasons:
1. To keep logic consistent, where we check if value is changed while updating any argument.
2. aws fsx update-file-system API is not working in below scenario: 
           when we are passing 2 parameters to LustreConfiguration and one of the parameter value do not have any difference from old value. UpdateFileSystem is working for LustreConfiguration only when we pass all parameters which needs to be updated with new value.